docs: update docs with v1.0.0 changes (#47)
project-stacker/project-stacker.github.io: docs/reference/stacker_file.md

`scratch`: which is an empty rootfs and can be used to host statically built binaries.

### `imports`

The `imports` directive describes what files should be made available in `/stacker/imports` during the `run` phase. There are three forms of importing supported today:

    /path/to/file

Will import a file or directory from the local filesystem. If the file or directory changes between stacker builds, it will be hashed and the new file will be imported on subsequent builds.

    http://example.com/foo.tar.gz

Will import foo.tar.gz and make it available in `/stacker`. Note that stacker will NOT update this file unless the cache is cleared, to avoid excess network usage. That means that updates after the first time stacker downloads the file will not be reflected.

    stacker://$name/path/to/file

> Will grab /path/to/file from the previously built layer `$name`.

Reviewer: we should replace 'layer' with 'image' here, even though it's not a new change

Author: Done

        - /foo/bar -> /bar/baz
	- /zomg

The first binds /foo/bar to /bar/baz, and the second binds host /zomg to
container /zomg.

At this time there is no awareness of change for any of these bind mounts, so
`--no-cache` should be used to re-build if the content of the bind mount has
changed.

### `config`

The `config` key is a special type of entry in the root of the `stacker.yaml` file.

> It cannot contain a layer definition; it is used to provide configuration

Reviewer: ```suggestion
It cannot contain an image definition; it is used to provide configuration
```

Author: Done

> applicable for building all the layers defined in this file. For example,

Reviewer: layers -> images

Author: Done

If the `prerequisites` list is present under the `config` key, stacker will

> make sure to build all the layers in the stacker.yaml files found at the paths

Reviewer: layers ->images

Author: Done
